### DISCUSSION TITLE
script, how to read BLE sensor (ESP32)

### DISCUSSION THREAD

**Discussion by jakrzy** [arendst/Tasmota#15377] [Q&A] [answered]
Hi to all.
How can I read the value of a Bluethooth sensor and assign it to a variable in a script. ?
I have a Mi Flora sensor, tasmota is visible, it also announces correctly after MQTT. However, using the tasmota script I cannot read this sensor and assign it to a variable.
e.g;
">T"
temp = Flora6cb47c # Temperature
Doesn't download data.
It is similar to the console trying to read:
01: 20: 13.823 CMD: Flora6cb47c # Temperature
01: 20: 13.839 MQT: stat / tasmota_799DCC / RESULT = {"Command": "Unknown"}

In the case of sensors connected directly to the tasmota, such as DHT, there is no problem with the reading.

[Tasmota 11.1.0 by Theo Arends](https://bit.ly/tasmota) 

Regards Jacek

**Comment by barbudor** [comment, marked as answer]
What are your Mi32Options settings (for 0, 1, 2 and 6) ?
What is the sensor message that is published ?

**Reply by barbudor** [reply]
If you are using `Mi32Option6 1` this is the reason for the script not to get the event
I have just pushed a PR https://github.com/arendst/Tasmota/pull/15383 which would solve that with `Mi32Option6 2`

**Reply by jakrzy** [reply]
Thank you, it helped.
I compiled the latest version of tasmota development with scripting support, turned on Mi32Option6 2 and in the script I can assign to variables, miflora sensor values.

Regards Jacek
![Clipboard01](https://user-images.githubusercontent.com/60155974/163728680-c9e63471-c242-48b2-85cf-40e8e00888ce.jpg)
k

**Reply by barbudor** [reply]
Pleased to have been able to help
